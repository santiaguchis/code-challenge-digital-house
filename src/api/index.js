export * from './apiRest';
export * from './services';