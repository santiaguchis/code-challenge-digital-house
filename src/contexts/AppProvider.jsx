import { useState } from 'react';
import { AppContext } from './AppContext';

const initialState = {
  user: {
    firstName: 'Ruben',
    lastName: 'Rodriguez'
  }
}

export const AppProvider = ({ children }) => {
  const [ movements, setMovements ] = useState([]);
  const [ filtered, setFiltered ] = useState([]);
  const [ totalPoints, setTotalPoints ] = useState(0);
  const [ filterSelected, setFilterSelected ] = useState('');

  const [ isLoading, setIsLoading ] = useState(true);

  const onHandleFilter = (items) => {
    setIsLoading(false);
    setFiltered(items)
  }
  const appState = {
    ...initialState,
    isLoading,
    movements,
    filtered,
    totalPoints,
    filterSelected,
    setMovements,
    setFiltered,
    setTotalPoints,
    setFilterSelected,
    setIsLoading,
    onHandleFilter,
  }
  return (
    <AppContext.Provider
      value={appState}
    >
      { children }
    </AppContext.Provider>
  )
}