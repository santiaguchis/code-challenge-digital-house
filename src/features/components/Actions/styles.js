import { StyleSheet } from 'react-native';

import { theme } from 'src/core/utils/styles';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
  }
})

export default styles;