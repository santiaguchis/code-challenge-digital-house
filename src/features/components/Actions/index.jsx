import { useContext } from 'react';
import { View } from 'react-native';

import { AppContext } from 'src/contexts';
import { Button, Separator } from 'src/core/components';
import { useFiltered } from 'src/hooks/useFiltered';

import styles from './styles';

const {
  container,
} = styles;

export const Actions = () => {
  const { 
    filterSelected, 
    setFilterSelected,
  } = useContext( AppContext );
  const { loadFilter } = useFiltered()
  
  const handleFiltered = ( filter ) => {
    setFilterSelected(filter)
    loadFilter(filter)
  }
  return (
    <View style={ container }>
      {
        filterSelected === 'all'
        ? <>
          <Button onPress={ () => handleFiltered('won') }>Ganados</Button>
          <Separator width={ 20 }/>
          <Button onPress={ () => handleFiltered('redeemed') }>Canjeados</Button>
        </>
        : <Button onPress={ () => handleFiltered('all') } size="lg">Todos</Button>
      }
    </View>
  )
}

export default Actions;