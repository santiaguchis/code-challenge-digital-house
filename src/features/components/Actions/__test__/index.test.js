import { render, screen } from '@testing-library/react-native';
import { AppProvider } from 'src/contexts';

import Actions from 'src/features/components/Actions';

let component;
describe('<Actions /> features/component', () => {
  const findTextButton = 'Todos'
  beforeEach(() => {
    component = render(
      <AppProvider>
        <Actions />
      </AppProvider>
    )
  })

  it('renderer element', () => {
    expect(component).toBeDefined();
  })

  it('Check text in component', () => {
    const textInComponent = screen.getByText(findTextButton);
    expect(textInComponent).toHaveTextContent(findTextButton);
  })

  it('Check snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  })
})