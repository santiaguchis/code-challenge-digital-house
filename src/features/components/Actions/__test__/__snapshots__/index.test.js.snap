// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actions /> features/component Check snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#334FFA",
        "borderRadius": 10,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 16,
        }
      }
      testID="text-button"
    >
      Todos
    </Text>
  </View>
</View>
`;
